samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -312,10 +317,22 @@ class TaskInstance(
 
     val commitStartNs = System.nanoTime()
     // first check if there were any unrecoverable errors during the async stage of the pending commit
-    // and if so, shut down the container.
+    // If there is unrecoverable error, increment the metric and the counter.
+    // Shutdown the container in the following scenarios:
+    // 1. skipCommitDuringFailureEnabled is not enabled
+    // 2. skipCommitDuringFailureEnabled is enabled but the number of exceptions exceeded the max count
+    // Otherwise, ignore the exception.
     if (commitException.get() != null) {
-      throw new SamzaException("Unrecoverable error during pending commit for taskName: %s." format taskName,
-        commitException.get())
+      metrics.commitExceptions.inc()
+      commitExceptionCounter += 1
+      if (!skipCommitDuringFailureEnabled || commitExceptionCounter > skipCommitExceptionMaxLimit) {
+        throw new SamzaException("Unrecoverable error during pending commit for taskName: %s. Exception Counter: %s"
+          format (taskName, commitExceptionCounter), commitException.get())
+      } else {
+        warn("Ignored the commit failure for taskName %s: %s" format (taskName, commitException.get().getMessage))
+        commitException.set(null)
+        commitInProgress.release()
```

Comment:
What are we releasing the semaphore here? I don't think this is necessary. 
It will always be released as a part of `finally` clause in the `handleCompletion` private method.